sos/plugins/__init__.py

```diff
@@ -829,13 +839,31 @@ def getmtime(path):
                     self._log_debug("skipping forbidden path '%s'" % _file)
                     continue
                 try:
-                    current_size += os.stat(_file)[stat.ST_SIZE]
+                    filestat = os.stat(_file)
                 except OSError:
                     self._log_info("failed to stat '%s'" % _file)
+                    continue
+                current_size += filestat[stat.ST_SIZE]
+                filetime = datetime.fromtimestamp(filestat[stat.ST_MTIME])
```

Comment:
Can't this be merged with `files.sort(key=getmtime, reverse=True)` called some lines above? Or maybe before this? Such that:

- we first cut off files that are too old to `--since` option
- then we sort the remaining ones per the same `mtime` parameter

Maybe we can also prevent the double-calling of `stat` to get `mtime`, then.

Reply:
I've moved everything before the loop and run a filter on the files list instead of filtering while looping